lib/wallet/txdb.js

```diff
@@ -1503,6 +1570,89 @@ class TXDB {
     });
   }
 
+  /**
+   * Get all transactions in ascending or decending order
+   * limited at a max of 100 transactions.
+   * @param {Number} acct
+   * @param {Number} limit
+   * @param {Boolean} reverse
+   * @returns {Promise} - Returns {@link TX}[].
+   */
+
+  async listHistory(acct, limit, reverse) {
+    assert(typeof acct === 'number');
+    assert(typeof limit === 'number');
+    assert(typeof reverse === 'boolean');
+
+    if (limit > 100)
+      throw new Error('Limit exceeds max of 100.')
+
+    const hashes = await this.bucket.values({
+      gte: layout.x.min(acct),
+      lte: layout.x.max(acct),
+      limit: limit,
+      reverse: reverse // TODO: Verify that `bdb` supports `reverse`
+    });
+
+    return Promise.all(hashes.map(async (hash) => await this.getTX(hash)));
+  }
+
+  /**
+   * Get the count of a transaction.
+   * @param {Number} acct
+   * @param {Buffer} txid
+   * @returns {Promise} - Returns Number.
+   */
+
+  async getCountForTX(acct, txid) {
+    assert(typeof acct === 'number');
+    assert(Buffer.isBuffer(txid));
+
+    const raw = await this.bucket.get(layout.w.encode(acct, txid));
+    if (!raw)
+      throw new Error('Transaction count not found.');
+    return raw.readUInt32BE(0, true);
+  }
+
+  /**
+   * Get all transactions in ascending or decending order
+   * after a txid and limited at a max of 100 transactions.
```

Comment:
It's exclusive, otherwise there would be overlap, there is a test case that asserts that there isn't.

Reply:
What's the use case for using "from" instead of "after", should that also be supported? For the case with pagination, getting the next page grabs the last txid from the previous result and passing it to the next call to make sure that even if the counts shift with a new block, the results will be continuous. Since there can be reorgs these guarantees will only be for blocks with 6 or more confirmations.

Reply:
Also "from" is an argument for `listtransactions` and it's a number there. Which reminds me, those methods should be updated to use the index also (but will have less functionality).

Reply:
I think the only argument I could think of is just expectations of usage. If it adds complexity then I think it's fine to not worry about "from" especially if it's very explicit that it's exclusive.

Reply:
Sounds good, added exclusive to the docs.